### ISSUE TITLE
[DOM-first] Fix `org.eclipse.jdt.core.tests.model.ReconcilerTests*` test suites

### ISSUE THREAD

**Issue by datho7561** [eclipse-jdtls/eclipse-jdt-core-incubator#83]
See https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/issues/12, treat this as a subissue for that epic

**Comment by mickaelistria** [comment]
Some of the test can be made to work by removing
```
project15.setOption(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES, JavaCore.ENABLED);
project15.setOption(JavaCore.COMPILER_PB_REPORT_PREVIEW_FEATURES, JavaCore.IGNORE);
```
The reason is that ASTParser runs some preliminary steps that cause more errors to be detected and reported; including the error report about preview enabled with an older compliance.
The fix for it is most likely best in JDT upstream.

**Comment by mickaelistria** [comment]
was fixed with #226
